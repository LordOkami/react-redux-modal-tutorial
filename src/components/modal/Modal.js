import React from "react";
import PropTypes from "prop-types";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";

import { openModal } from "../../routines";

import OverlayModal from "./OverlayModal";

export const Modal = ({ openModal, modal }) => (
  <React.Fragment>
    <div className="container">
      {console.log(modal)}
      <button className="centered" onClick={openModal}>
        Open Modal
      </button>
    </div>
    {modal.open && <OverlayModal />}
  </React.Fragment>
);

Modal.propTypes = {
  openModal: PropTypes.func,
  modal: PropTypes.object
};

const mapDispatchToProps = dispatch => ({
  ...bindActionCreators({ openModal }, dispatch)
});

const mapStateToProps = ({ modal }) => ({
  modal
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Modal);
