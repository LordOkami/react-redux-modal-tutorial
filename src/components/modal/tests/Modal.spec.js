import React from "react";
import { shallow } from "enzyme";
import { Modal } from "../Modal";

describe("<Modal />", () => {
  const modal = { open: false };
  const openModal = jest.fn;
  const props = { modal, openModal };
  it("matches the latest snapshot", () => {
    const wrapper = shallow(<Modal {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
