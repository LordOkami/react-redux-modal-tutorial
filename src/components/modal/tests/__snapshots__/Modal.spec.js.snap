// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> matches the latest snapshot 1`] = `
<Fragment>
  <div
    className="container"
  >
    <button
      className="centered"
      onClick={[Function]}
    >
      Open Modal
    </button>
  </div>
</Fragment>
`;
